utils/syncpool_test.go

```diff
@@ -0,0 +1,90 @@
+package utils
+
+import (
+	"testing"
+)
+
+func TestNewPool(t *testing.T) {
+	newFunc := func() int {
+		return 42
+	}
+	resetFunc := func(x int) int {
+		return x + 1
+	}
+	pool := NewPool(newFunc, resetFunc, nil)
+
+	if pool == nil {
+		t.Fatal("Expected pool to be initialized")
+	}
+
+	x := pool.Get()
+	if x != 42 {
+		t.Errorf("Expected Get() to return 42, got %d", x)
+	}
+}
+
+func TestPool_GetPut(t *testing.T) {
+	pool := NewPool(func() string { return "initial" }, nil, nil)
+
+	// Get a new item
+	s := pool.Get()
+	if s != "initial" {
+		t.Errorf("Expected 'initial', got '%s'", s)
+	}
+
+	// Put back a different string
+	pool.Put("updated")
+
+	// Get should return the updated string
+	s = pool.Get()
+	if s != "updated" {
+		t.Errorf("Expected 'updated', got '%s'", s)
+	}
+}
+
+func TestNewBytesPool(t *testing.T) {
+	size := 1024
+	bytesPool := NewBytesPool(size, 0)
+	if bytesPool == nil {
+		t.Fatal("Expected bytes pool to be initialized")
```

Comment:
can you use the https://github.com/stretchr/testify/require package to reduce the size of these assertion blocks?